languages/csharp/src/main/kotlin/io/vrap/codegen/languages/csharp/extensions/CsharpVrapExtensions.kt

```diff
@@ -0,0 +1,119 @@
+package io.vrap.codegen.languages.csharp.extensions
+
+import io.vrap.rmf.codegen.types.*
+import io.vrap.rmf.raml.model.util.StringCaseFormat
+import com.hypertino.inflector.English
+
+fun VrapType.simpleName(): String {
+    return when (this) {
+        is VrapScalarType -> this.scalarType
+        is VrapDateTimeType -> this.simpleClassName
+        is VrapEnumType -> this.simpleClassName
+        is VrapObjectType -> if(this.simpleClassName == "DateTime" || this.simpleClassName == "Object") this.simpleClassName else "I${this.simpleClassName}"
+        is VrapAnyType -> this.baseType
+        is VrapArrayType -> """List\<${this.itemType.simpleName()}\>"""
+        is VrapNilType -> throw IllegalStateException("$this has no simple class name.") as Throwable
+    }
+}
+
+fun VrapType.toCsharpVType(): VrapType {
+    return when (this) {
+        is VrapObjectType -> {
+            VrapObjectType(`package` = this.`package`.toCsharpPackage(), simpleClassName = this.simpleClassName)
+        }
+        is VrapEnumType -> {
+            VrapEnumType(`package` = this.`package`.toCsharpPackage(), simpleClassName = this.simpleClassName)
+        }
+        is VrapArrayType -> {
+            VrapArrayType(this.itemType.toCsharpVType())
+        }
+        else -> {
+            this
+        }
+
+    }
+}
+
+fun VrapType.csharpPackage(): String {
+    var packageName = ""
+    if (this is VrapObjectType) packageName = this.`package`
+    else if (this is VrapEnumType) packageName = this.`package`
+
+    if(!packageName.isNullOrEmpty())
+        packageName = packageName.toCsharpPackage()
+    return packageName
+}
+//need refactor
+fun VrapType.requestBuildersPackage(resourceNamePlural: String): String {
+    var cPackage = this.csharpPackage()
+    return cPackage.replace("Clientss","Client").replace("Clients","Client")+".RequestBuilders"+"."+resourceNamePlural
```

Comment:
Is there really something with `Clientss` existing?

Reply:
I just refactored it.